ssd1306/ssd1306.c

```diff
@@ -189,12 +176,26 @@ void ssd1306_UpdateScreen(void) {
     //  * 32px   ==  4 pages
     //  * 64px   ==  8 pages
     //  * 128px  ==  16 pages
-    for(uint8_t i = 0; i < SSD1306_HEIGHT/8; i++) {
-        ssd1306_WriteCommand(0xB0 + i); // Set the current RAM page address.
-        ssd1306_WriteCommand(0x00);
-        ssd1306_WriteCommand(0x10);
-        ssd1306_WriteData(&SSD1306_Buffer[SSD1306_WIDTH*i],SSD1306_WIDTH);
-    }
+#if defined(SSD1306_USE_SPI)
+	ssd1306_WriteCommand(0xB0); // Set the current RAM page address.
+    ssd1306_WriteCommand(0x00);
+    ssd1306_WriteCommand(0x10);
+	HAL_GPIO_WritePin(SSD1306_CS_Port, SSD1306_CS_Pin, GPIO_PIN_RESET); // select OLED
+    HAL_GPIO_WritePin(SSD1306_DC_Port, SSD1306_DC_Pin, GPIO_PIN_SET); // data
+#if defined(SSD1306_USE_DMA)
+    HAL_SPI_Transmit_DMA(&SSD1306_SPI_PORT, SSD1306_Buffer, SSD1306_BUFFER_SIZE);
+#else
+    HAL_SPI_Transmit(&SSD1306_SPI_PORT, SSD1306_Buffer, SSD1306_BUFFER_SIZE, HAL_MAX_DELAY);
```

Comment:
What are the benefits of using HAL_SPI_Transmit_DMA vs HAL_SPI_Transmit here? Isn't the main advantage of DMA that you can transmit data asynchronously? This being said I don't have much experience with DMA...

Reply:
Yes, DMA can transfer data asynchronously, and transfers much faster than without DMA. Using DMA also makes CPU not wasting time on moving data, so CPU have more time to do other work.